src/python/pants/backend/native/tasks/native_compile.py

```diff
@@ -211,18 +224,17 @@ def _compile(self, compile_request):
     NB: This method must arrange the output files so that `collect_cached_objects()` can collect all
     of the results (or vice versa)!
     """
-    sources = compile_request.sources
+    sources, _ = self._split_sources_and_headers(
```

Comment:
Since headers are always thrown away, consider switching to a generator and maybe renaming such that:
```python
sources_minus_headers = list(self._iter_sources_minus_headers(compile_request))
```

Reply:
I'm curious - what does making this a generator and casting as a list net here?

Reply:
The splitting headers and sources method is absolutely an artifact of a previous implementation that got rewritten just before it was merged, where the headers were used -- this is no longer necessary (and never was) so I think the edit is potentially in the vein of just not introducing the sources to throw away.

Reply:
Under the assumption in my above comment, I have make the change here, and it looks great.